rq/queue.py

```diff
@@ -663,11 +663,23 @@ def enqueue_call(
             on_success=on_success,
             on_failure=on_failure,
         )
+        return self._check_dependencies(job, pipeline=pipeline, at_front=at_front)
 
+    def _check_dependencies(self, job: 'Job', pipeline: Optional['Pipeline'] = None, at_front: bool = False) -> Job:
```

Comment:
`_check_dependencies` shouldn't enqueue job to the queue. Besides, we already have `setup_dependencies`.

Reply:
I changed the name from `_check_dependencies` to `_enqueue_checking_dependencies` to be more clear.

Reply:
Deleted via https://github.com/rq/rq/compare/3f5e188fb93be91e9c7452d1a5724cdff9d5096a..4cf0db1ea9f2c93bbd2557882438efff3d2b77de